datagovharvester/extract/dcatus.py

```diff
@@ -0,0 +1,33 @@
+from .utils import download_catalog
+import json
+from datagovharvester.utils.s3_utilities import (
+    upload_to_S3,
+    create_s3_payload,
+)
+
+
+def extract_json_catalog(url, job_id, S3_client, bucket_name):
+    errors = []
+
+    try:
+        data = download_catalog(url)
+    except Exception as e:
+        # do something with the error
+        return e
+
+    if not data.get("dataset", None):
+        # do something with the error
+        return "no dataset key"
+
+    for idx, record in enumerate(data["dataset"]):
+        try:
+            record = json.dumps(record)
+            key_name = f"{job_id}_{idx}_extract.json"
```

Comment:
I would actually have this as `source_id/job_id/idx.json`. This allows for easier browsing by humans, if we have to debug manually.

Reply:
updated [here](https://github.com/GSA/datagov-harvesting-logic/commit/41209f80fc7566d403ccb7400aade30b7f0ac551) and corresponding location in store_record_in_s3 function.